frame/bonded-finance/src/tests.rs

```diff
@@ -92,6 +95,46 @@ proptest! {
 			  })?;
 	  }
 
+	  #[test]
+	  fn generate_vesting_schedule(offer in simple_offer(1)) {
+		ExtBuilder::build().execute_with(|| {
+			let mut offer = offer.clone();
+			offer.beneficiary = BOB;
+			System::set_block_number(1);
+			prop_assert_ok!(Tokens::mint_into(NATIVE_CURRENCY_ID, &ALICE, Stake::get()));
+			prop_assert_ok!(Tokens::mint_into(offer.reward.asset, &ALICE, offer.reward.amount));
+			let offer_id = BondedFinance::do_offer(&ALICE, offer.clone(), false);
+			prop_assert_ok!(offer_id);
+			let offer_id = offer_id.expect("impossible; qed");
+
+			prop_assert_ok!(BondedFinance::do_bond(offer_id, &BOB, 1, false));
```

Comment:
why you test bond offer created in vesting schdule created test?

Reply:
Yeah this is not nice, we can just test even publishing in the vesting pallet

Reply:
The `vesting` pallet already has a test for `VestingScheduleAdded` which was passing, even though the event was not generated when bonding. Doesn't it make sense that we make sure in the bonded finance tests that this event is generated, or should we keep it strictly decoupled?

Reply:
I guess the test you would want to do is a test that makes sure that 
```
<Self as VestedTransfer>::vested_transfer(asset, &from, &to, schedule)?;
```
emits the event in vesting pallet. That should work for all pallets who would depend on `VestedTransfer` trait, so that test naturally belongs in the vesting pallet.

Reply:
fixed ✅